File: work-spring/BoardOAuth/src/main/java/com/grepp/boardjwt/model/repository/UserRepository.java
package com.grepp.boardjwt.model.repository;


import com.grepp.boardjwt.model.entity.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends JpaRepository<UserEntity, String> {

  UserEntity findByUsername(String username);
  Boolean existsByUsername(String username);
  UserEntity findByUsernameAndPassword(String username, String password);
}

File: work-kotlin/day1021_todo/src/main/kotlin/com/grepp/todo/model/repository/UserRepository.kt
package com.grepp.todo.model.repository

import com.grepp.todo.model.entity.UserEntity
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : CrudRepository<UserEntity, Long> {
    // 회원가입 함수는 자동으로 create 만들어지니까.
    // 로그인 함수만 만들어놓을까요 : 로그인 실패시 결과 null 허용되게 ? 붙여놨습니당
    fun findByUsernameAndPassword(username:String, password:String): UserEntity?// login할 때 함수
}